packages/amplication-server/src/core/resource/resource.service.ts

```diff
@@ -1552,10 +1552,28 @@ export class ResourceService {
   }
 
   async resources(args: FindManyResourceArgs): Promise<Resource[]> {
+    const { serviceTemplateId, ...where } = args.where;
+    delete args.where?.serviceTemplateId;
+
+    let resourceIds: string[] = undefined;
+    if (serviceTemplateId) {
+      if (!args.where?.project?.id) {
+        throw new Error(
+          "project.id is required when searching by serviceTemplateId"
+        );
+      }
+
+      resourceIds = await this.serviceSettingsService.getServiceIdsByTemplateId(
+        args.where.project.id,
+        serviceTemplateId
+      );
+    }
+
     return this.prisma.resource.findMany({
       ...args,
       where: {
-        ...args.where,
+        ...where,
+        id: resourceIds ? { in: resourceIds } : where.id,
```

Comment:
This method should return multiple resources, so adding `id: where.id` (in the case that `serviceTemplateId` was not specified) doesn't make sense to me

Reply:
the idea is that we override the provided value for where. id in case we want to filter specific resources by id, otherwise we use the original value of where.id which probably is undefined but we dont really know or care